src/main/java/org/ow2/proactive/catalog/repository/CatalogObjectGrantRepository.java

```diff
@@ -46,32 +43,45 @@
 
-    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.grantee = ?1 AND coge.catalogObjectRevisionEntity.id = ?2 AND coge.bucketEntity.id=?3 And coge.granteeType='user'")
-    List<CatalogObjectGrantEntity> findAllCatalogObjectGrantsAssignedToAUsername(String username, long catalogObjectId,
+    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.catalogObjectRevisionEntity.id = ?1 AND coge.grantee = ?2 AND coge.bucketEntity.id=?3 AND coge.granteeType='group'")
+    CatalogObjectGrantEntity findCatalogObjectGrantByUserGroupForUpdate(long catalogObjectId, String userGroup,
             long bucketId);
 
-    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.grantee = ?1 AND coge.catalogObjectRevisionEntity.id = ?2 AND coge.bucketEntity.id=?3 And coge.granteeType='group'")
-    List<CatalogObjectGrantEntity> findAllCatalogObjectGrantsAssignedToAUserGroup(String userGroup,
-            long catalogObjectId, long bucketId);
-
-    @Query(value = "SELECT coge.bucketEntity.id FROM CatalogObjectGrantEntity coge WHERE coge.grantee = ?1 AND coge.granteeType='user'")
+    @Query(value = "SELECT coge.bucketEntity.id FROM CatalogObjectGrantEntity coge WHERE coge.grantee = ?1 AND coge.granteeType='user' AND coge.accessType<>'noAccess'")
     List<Long> findAllBucketsIdFromCatalogObjectGrantsAssignedToAUsername(String username);
 
-    @Query(value = "SELECT coge.bucketEntity.id FROM CatalogObjectGrantEntity coge WHERE coge.grantee in ?1 AND coge.granteeType='group'")
+    @Query(value = "SELECT coge.bucketEntity.id FROM CatalogObjectGrantEntity coge WHERE coge.grantee in ?1 AND coge.granteeType='group' AND coge.accessType<>'noAccess'")
     List<Long> findAllBucketsIdFromCatalogObjectGrantsAssignedToAUserGroup(List<String> userGroups);
 
-    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.grantee = ?1 AND coge.granteeType='user'")
-    List<CatalogObjectGrantEntity> findAllGrantsAssignedToAUsername(String username);
-
-    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.grantee in ?1 AND coge.granteeType='group'")
-    List<CatalogObjectGrantEntity> findAllGrantsAssignedToAUserGroup(List<String> userGroups);
-
     List<CatalogObjectGrantEntity> deleteAllByBucketEntityIdAndCatalogObjectRevisionEntityId(long bucketId,
             long catalogObjectId);
+
+    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.grantee = ?1 AND coge.granteeType='user' AND coge.accessType='noAccess'")
+    CatalogObjectGrantEntity findAllNoAccessGrantsAssignedToAUsername(String username);
+
+    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.grantee in ?1 AND coge.granteeType='group' AND coge.accessType='noAccess'")
+    List<CatalogObjectGrantEntity> findAllNoAccessGrantsAssignedToAUserGroup(List<String> userGroups);
```

Comment:
Can you improve the naming coherence of these functions ?

Example:
findAllNoAccessGrantsAssignedToAUsername but findAllObjectGrantsAssignedToAUser

findAllNoAccessGrantsAssignedToAUserGroup but findAllObjectGrantsAssignedToUserGroups